cadc-vos/src/main/java/ca/nrc/cadc/vos/TransferReader.java

```diff
@@ -223,70 +208,67 @@ private Transfer parseTransfer(Document document, String targetScheme)
         if (keepBytesStr != null)
             keepBytes = keepBytesStr.equalsIgnoreCase("true");
 
-        Transfer ret = new Transfer(target, direction, view, protocols, keepBytes);
+        Transfer ret = new Transfer(direction);
+        ret.targets = targets;
```

Comment:
As mentioned in slack:  this should be `ret.getTargets().addAdd(targets)` 
So, please change Transfer so that `targets` and `protocols` are private and have getters and setters.